brayns/Brayns.cpp

```diff
@@ -267,47 +242,46 @@ struct Brayns::Impl
 
     Scene& getScene( )
     {
-        return *_scene;
+        return *_engine->getScene();
     }
 
     Camera& getCamera( )
     {
-        return *_camera;
+        return *_engine->getCamera();
     }
 
     FrameBuffer& getFrameBuffer( )
     {
-        return *_frameBuffer;
+        return *_engine->getFrameBuffer();
     }
 
 private:
     void _render( )
     {
-        const std::string& renderer =
-            _parametersManager->getRenderingParameters().getRenderer();
-        if( _activeRenderer != renderer )
-        {
-            _activeRenderer = renderer;
-#if(BRAYNS_USE_DEFLECT || BRAYNS_USE_REST)
-            _extensionParameters.renderer = _renderers[_activeRenderer];
-#endif
-        }
-        _renderers[_activeRenderer]->render( _frameBuffer );
+
+        _engine->setActiveRenderer(
+            _parametersManager->getRenderingParameters().getRenderer());
+        _engine->render();
     }
 
     void _setDefaultCamera()
     {
-        const Boxf worldBounds = _scene->getWorldBounds( );
+        ScenePtr scene = _engine->getScene();
+        CameraPtr camera = _engine->getCamera();
+        FrameBufferPtr frameBuffer = _engine->getFrameBuffer();
+        const Vector2i& frameSize = frameBuffer->getSize();
+
+        const Boxf worldBounds = scene->getWorldBounds( );
         const Vector3f target = worldBounds.getCenter( );
         const Vector3f diag   = worldBounds.getSize( );
         Vector3f position = target;
         position.z( ) -= diag.z( );
 
         const Vector3f up  = Vector3f(0.f,1.f,0.f);
-        _camera->setInitialState(position, target, up);
-        _camera->setAspectRatio(
-            static_cast< float >( _frameSize.x()) /
-            static_cast< float >( _frameSize.y()));
+        camera->setInitialState(position, target, up);
```

Comment:
There are multiple places the same code, can you gather into a function. Normalization means decreasing the repetition.